src/bench/bench_pocketcoin.cpp

```diff
@@ -11,91 +11,144 @@
+                     "Environment Variables:\n"
+                     "\n"
+                     "  To attach a profiler you can run a benchmark in endless mode. This can be\n"
+                     "  done with the environment variable NANOBENCH_ENDLESS. E.g. like so:\n"
+                     "\n"
+                     "    NANOBENCH_ENDLESS=MuHash ./bench_bitcoin -filter=MuHash\n"
+                     "\n"
+                     "  In rare cases it can be useful to suppress stability warnings. This can be\n"
+                     "  done with the environment variable NANOBENCH_SUPPRESS_WARNINGS, e.g:\n"
+                     "\n"
+                     "    NANOBENCH_SUPPRESS_WARNINGS=1 ./bench_bitcoin\n"
+                     "\n"
+                     "Notes:\n"
+                     "\n"
+                     "  1. pyperf\n"
+                     "     https://github.com/psf/pyperf\n"
+                     "\n"
+                     "  2. CPU pinning & isolation\n"
+                     "     https://pyperf.readthedocs.io/en/latest/system.html\n"
+                     "\n"
+                     "  3. nanobench\n"
+                     "     https://github.com/martinus/nanobench\n"
+                     "\n";
 
         return EXIT_SUCCESS;
     }
 
-    // Set the datadir after parsing the bench options
-    const fs::path bench_datadir{SetDataDir()};
-
     SHA256AutoDetect();
     RandomInit();
     ECC_Start();
-    SetupEnvironment();
+    globalVerifyHandle.reset(new ECCVerifyHandle());
+    //SetupEnvironment();
+    //SetupNetworking();
+    //InitSignatureCache();
+    //InitScriptExecutionCache();
 
-    int64_t evaluations = gArgs.GetArg("-evals", DEFAULT_BENCH_EVALUATIONS);
-    std::string regex_filter = gArgs.GetArg("-filter", DEFAULT_BENCH_FILTER);
-    std::string scaling_str = gArgs.GetArg("-scaling", DEFAULT_BENCH_SCALING);
-    bool is_list_only = gArgs.GetBoolArg("-list", false);
+    SelectParams(CBaseChainParams::REGTEST);
 
-    double scaling_factor;
-    if (!ParseDouble(scaling_str, &scaling_factor)) {
-        fprintf(stderr, "Error parsing scaling factor as double: %s\n", scaling_str.c_str());
-        return EXIT_FAILURE;
-    }
 
-    std::unique_ptr<benchmark::Printer> printer = MakeUnique<benchmark::ConsolePrinter>();
-    std::string printer_arg = gArgs.GetArg("-printer", DEFAULT_BENCH_PRINTER);
-    if ("plot" == printer_arg) {
-        printer.reset(new benchmark::PlotlyPrinter(
-            gArgs.GetArg("-plot-plotlyurl", DEFAULT_PLOT_PLOTLYURL),
-            gArgs.GetArg("-plot-width", DEFAULT_PLOT_WIDTH),
-            gArgs.GetArg("-plot-height", DEFAULT_PLOT_HEIGHT)));
-    }
+    PocketDb::InitSQLite(GetDataDir() / "pocketdb");
 
-    benchmark::BenchRunner::RunAll(*printer, evaluations, scaling_factor, regex_filter, is_list_only);
+    // Go up two directories to access the checkpoints folder, assume we are running in /src/test
+    fs::path checkpointsPath = fs::system_complete("../..");
```

Comment:
Why not use same logic as for other nets? I mean searching for checkpoints in datadir. This assuming prevents to run bench tests if build tree differ from source that is a common situation.

Reply:
We will deal with this issue here https://github.com/pocketnetteam/pocketnet.core/issues/182